src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -178,13 +179,77 @@ SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::stri
+    PowershellCoreProfileGenerator powerShellGenerator{};
+    _executeGenerator(powerShellGenerator);
+
+    WslDistroGenerator wslGenerator{};
+    _executeGenerator(wslGenerator);
+
+    AzureCloudShellGenerator acsGenerator{};
+    _executeGenerator(acsGenerator);
+
+    VisualStudioGenerator vsGenerator{};
+    _executeGenerator(vsGenerator);
+
 #if TIL_FEATURE_DYNAMICSSHPROFILES_ENABLED
-    _executeGenerator(SshHostGenerator{});
+    SshHostGenerator sshGenerator{};
+    _executeGenerator(sshGenerator);
 #endif
+
+    PowershellInstallationProfileGenerator pwshInstallationGenerator{};
+    _executeGenerator(pwshInstallationGenerator);
+
+    _cleanupPowerShellInstaller(!powerShellGenerator.GetPowerShellInstances().empty());
+}
+
+// Retrieve the "Install Latest PowerShell" profile and...
+// - add a comment to the JSON to indicate it's conditionally applied
+// - (if PowerShell is installed) mark it for deletion
+void SettingsLoader::_cleanupPowerShellInstaller(bool isPowerShellInstalled)
+{
+    const hstring pwshInstallerNamespace{ PowershellInstallationProfileGenerator::Namespace };
+    if (extensionPackageMap.contains(pwshInstallerNamespace))
+    {
+        if (const auto& fragExtList = extensionPackageMap[pwshInstallerNamespace]->Fragments(); fragExtList.Size() > 0)
+        {
+            Json::StreamWriterBuilder styledWriter;
+            styledWriter["indentation"] = "    ";
+            styledWriter["commentStyle"] = "All";
+
+            auto fragExt = get_self<FragmentSettings>(fragExtList.GetAt(0));
+
+            // We want the comment to be the first thing in the object,
+            // "closeOnExit" is the first property, so target that.
+            auto fragExtJson = _parseJSON(til::u16u8(fragExt->Json()));
+            fragExtJson[JsonKey(ProfilesKey)][0]["closeOnExit"].setComment(til::u16u8(fmt::format(FMT_COMPILE(L"// {}"), RS_(L"PowerShellInstallationProfileJsonComment"))), Json::CommentPlacement::commentBefore);
+            fragExt->Json(hstring{ til::u8u16(Json::writeString(styledWriter, fragExtJson)) });
+
+            if (const auto& profileEntryList = fragExt->NewProfilesView(); profileEntryList.Size() > 0)
+            {
+                auto profileEntry = get_self<FragmentProfileEntry>(profileEntryList.GetAt(0));
+
+                // We want the comment to be the first thing in the object,
+                // "closeOnExit" is the first property, so target that.
+                auto profileJson = _parseJSON(til::u16u8(profileEntry->Json()));
+                profileJson["closeOnExit"].setComment(til::u16u8(fmt::format(FMT_COMPILE(L"// {}"), RS_(L"PowerShellInstallationProfileJsonComment"))), Json::CommentPlacement::commentBefore);
+                profileEntry->Json(hstring{ til::u8u16(Json::writeString(styledWriter, profileJson)) });
+
+                // If PowerShell is installed, mark the installer profile for deletion
+                if (isPowerShellInstalled)

```

Comment:
Do we need to generate the installer profile at all then?

Reply:
Yes and no. Let me explain.

Profile generators leave a stub in the settings.json. Thus, the installer leaves this stub:
```js
{
    "guid": "{965a10f2-b0f2-55dc-a3c2-2ddbf639bf89}",
    "hidden": false,
    "name": "Install Latest PowerShell",
    "source": "Windows.Terminal.InstallPowerShell"
},
```
It's important that we go through the process of marking it for deletion so that this profile isn't left around for the user to clean up. That's why we have to mark it for deletion.

That said, you make a good point. If we already know that PowerShell is installed, let's just skip running the installer's generator at all.